test/triangular.jl

```diff
@@ -0,0 +1,65 @@
+@testset "Triangular-matrix multiplication" begin
```

Comment:
Worth adding an `srand` here, to keep testing consistent and predictable.

Generally I'd agree with @andyferris that hard coded test cases are better for basic unit tests (simply from a clarity point of view in being extremely explicit when something goes wrong).  However, you've got a lot of cases here, so hard coding them all doesn't seem practical.  (Generally I'd also aim to cover each code path only once or so for each type of input, just to keep testing time under control.  We do need a better way to do better comprehensive testing vs fast unit tests, but that's a problem for the package as a whole - I'd say we go with what's here for now :-) )